### PR TITLE
Use injector in the gmfPermalink service

### LINKED ISSUE #1962: Removes permalink required dependencies
On this file:
https://github.com/camptocamp/ngeo/blob/master/contribs/gmf/src/services/permalink.js

We should:
- Replace nearly all the ingection by the ingector.
- Get the "elements" from the ingector.
- Don't crash if an element isn't present.

### PR DESCRIPTION
**Pull request by ger-benjamin**
Fix #1962 

Question (from the issue):
> Don't crash if an element isn't present.

I can do that but it will crash later... What I'm supposed to do ? Adding if statements everywhere in this file ?

### CODE REVIEW

**Comment by sbrunner** [comment]
I think that we should use the has method on the injector to know if an element is present?

Than if an element isn't present the reference will be null, and we should add some if to don't fill and use the related part...

**Comment by ger-benjamin** [comment]
Done, thanks Sbrunner

File: contribs/gmf/src/services/permalink.js [hunk 1]
```diff
@@ -122,115 +123,109 @@ gmf.Permalink = function($timeout, ngeoBackgroundLayerMgr, ngeoDebounce,
...
-  /**
-   * @type {string}
-   * @private
-   */
-  this.defaultTheme_ = defaultTheme;
+    $injector.get('gmfThemeManager') : null;
 
   /**
-   * @type {angular.Scope}
+   * @type {?string}
```
Comment (sbrunner) [inline contribs/gmf/src/services/permalink.js]: Prefer `{string|undefined}` for native types (string, number and boolean) :-)

File: contribs/gmf/src/services/permalink.js [hunk 2]
```diff
@@ -73,36 +72,38 @@ gmf.module.value('gmfPermalinkOptions',
  * - whether to add a crosshair feature in the map or not
  * - the dimensions value
  *
+ * To avail the whole possibilities offer by the permalink, these services
```
Comment (fredj) [inline contribs/gmf/src/services/permalink.js]: `To have` instead of `To avail` ?

**Pull request merged**